### PR TITLE
Backend authentication

### PR DESCRIPTION
**Pull request by Unicus-Tom**

### CODE REVIEW

File: RestApi/Sepes.RestApi/appsettings.Development.json [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
       "System": "Information",
       "Microsoft": "Information"
     }
+  },
+  "Jwt": {
+    "Key": "secretkey2asfafds56865",
+    "Issuer": "example.com"
+  },
+  "AzureLogToken": {
+    "ServiceApiKey": "AZURETokenHere"
```
Comment (gjersvik) [inline RestApi/Sepes.RestApi/appsettings.Development.json]: You should not need to have any config values here that are the same as in appsettings.json

File: RestApi/Sepes.RestApi/Startup.cs [hunk 2]
```diff
@@ -30,10 +42,44 @@ public void ConfigureServices(IServiceCollection services)
...
+            services.AddApplicationInsightsTelemetry(Configuration["AzureLogToken:ServiceApiKey"]);
+            services.Configure<AppSettings>(Configuration.GetSection("Jwt"));
+            services.AddOptions();
+
+            services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)  
+            .AddJwtBearer(options =>  
+            {  
+            options.TokenValidationParameters = new TokenValidationParameters  
+                {  
+                ValidateIssuer = false,  //TODO set to true before MVP
```
Comment (gjersvik) [inline RestApi/Sepes.RestApi/Startup.cs]: Small tings like this that are easy to forget should realy be GitHub issues.

**Review by gjersvik — APPROVED** [review]
I would recommenced turning some of these todos into GitHub issues.

File: RestApi/Sepes.RestApi/Controllers/AuthController.cs [hunk 3]
```diff
@@ -13,6 +23,102 @@ namespace Sepes.RestApi.Controller
...
+        public string idToken { get; set; }
+        public string Expiration { get; set; }
+    }
+
+       public class SepesTokenClass{
+        //Judge what parameters are needed for just refreshing. Username and such should be in token content body
+        public string Username { get; set; }
+        public string idToken { get; set; }
+        public string Expiration { get; set; }
     }
```
Comment (Kris2ffer) [inline RestApi/Sepes.RestApi/Controllers/AuthController.cs:116]: Would like to see one generic token class instead of two that are basically the same.

**Review by Kris2ffer — APPROVED** [review]

File: RestApi/Sepes.RestApi/Controllers/AuthController.cs (hunk 3 above)
Comment (Unicus-Tom) [reply RestApi/Sepes.RestApi/Controllers/AuthController.cs:116]: They represent two different token formats. Currently contain the same content but may differ in the future. If it turns out we dont need two different classes it is trivial to combine them later.

File: RestApi/Sepes.RestApi/Startup.cs (hunk 2 above)
Comment (Unicus-Tom) [reply RestApi/Sepes.RestApi/Startup.cs]: Sure, I will get it done before my last commit

**Pull request merged**